deepspeed/runtime/pipe/schedule.py

```diff
@@ -243,7 +243,7 @@ def steps(self):
     def num_pipe_buffers(self):
         """As many buffers as the distance from this stage to the last stage.
         """
```

Comment:
```suggestion
        """Return the number of pipeline buffers required for this stage.
        
        This is equivalent to the maximum number of in-flight forward passes,
        since we need to remember the activations of forward passes in order
        to run backpropagation. For synchronous 1F1B, this is equivalent to
        the index difference between this stage and the last stage.
        """
```
Just thought this was a good chance to improve explanation. Also now that we removed the `+1`, the word "distance" no longer makes sense, at least in the sense the original code writer intended.

Reply:
Totally agreed, this is a great suggestion.